cmd/ctr/commands/content/content.go

```diff
@@ -536,17 +537,16 @@ func edit(context *cli.Context, rd io.Reader) (io.ReadCloser, error) {
 	cmd.Env = os.Environ()
 
 	if err := cmd.Run(); err != nil {
-		tmp.Close()
```

Comment:
newbie question: Dont we need to remove the temporary file that was created if the code errors out? Currently the file is removed only if the edit is success right?

Reply:
yes. all error paths after the temp file has been created should return an `onCloser` to delete the created file.

Reply:
I see that you have already made the change. May be @fuweid can give more info, whether `onCloser` is always required.

Reply:
@fuweid ,could you take a look, i have do some very simple test, havn't found left undeleted tmp files.

Reply:
basically, we don't use instance if the func returns error.
I think we can use defer to handle cleanup, like:

```go
func edit(context *cli.Context, rd io.Reader) (_ io.ReadCloser, retErr error) {
     ...
     tmp, err := os.CreateTemp(os.Getenv("XDG_RUNTIME_DIR"), "edit-")
     if err != nil {
	return nil, err
     }
     defer func() {
           if retErr != nil {
                os.Remove(tmp.Name())
           }
     }()
    ...
}
```